src/pygeom.c

```diff
@@ -76,6 +76,69 @@ static PyObject *GeometryObject_str(GeometryObject *self)
     return GeometryObject_ToWKT(self, "%s");
 }
 
+static Py_hash_t GeometryObject_hash(GeometryObject *self)
+{
+    void *context = geos_context[0];
+    unsigned char *wkb;
+    size_t size;
+    Py_hash_t x;
+
+    if (self->ptr == NULL) {
+        return -1;
+    }
+    GEOSWKBWriter *writer = GEOSWKBWriter_create_r(context);
```

Comment:
@caspervdw what about using [GEOSGeomToWKB_buf_r](https://github.com/libgeos/geos/blob/3.8/capi/geos_ts_c.cpp#L1370)?  It seems like that would avoid needing to manage the writer yourself.

Example [usage](https://github.com/libgeos/geos/blob/3.8/tests/thread/threadtest.c#L222-L223).

Reply:
According to the C API header that is function is deprecated:

https://github.com/libgeos/geos/blob/c01c8c1c30091d655ba888790c2daeec1f33bae3/capi/geos_c.h.in#L220

(but not sure how/if this is enforced)

Reply:
I think it is no good to use deprecated functions.